### PR TITLE
Misc test and build clean-up

### PR DESCRIPTION
**Pull request by ayj**
**What this PR does / why we need it**:

* Fix the following bazel WARNING

```
WARNING: <...> /external/bazel_tools/tools/build_defs/pkg/pkg.bzl:196:9: pkg_tar: renaming non-dict `files` attribute to `srcs
```

* Enable consistent checks in bash scripts (errexit, nounset, pipefail)

* Remove noisy debug print fro bin/bazel_to_go

* Remove bash color codes (i.e. "\033[1;34m"). This makes visually parsing prow test logs more difficult. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by rshriram** [comment]
@ayj let's test one at a time.

**Comment by rshriram** [comment]
Please test the defaults pr after this

**Comment by ayj** [comment]
SGTM - i'll hold off on rebase/resting until https://github.com/istio/pilot/pull/1243 is merged

File: tools/deb/BUILD [hunk 1]
```diff
@@ -2,7 +2,7 @@ load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar", "pkg_deb")
 
 pkg_tar(
     name = "agent-bin",
-    files = [
+    srcs = [
```
Comment (ldemailly) [inline tools/deb/BUILD]: revert this ?
```
W0915 01:32:57.236] ERROR: /home/bootstrap/go/src/github.com/istio/pilot/tools/deb/BUILD:3:1: //tools/deb:agent-bin: no such attribute 'srcs' in 'pkg_tar' rule.
```

Comment (ayj) [reply tools/deb/BUILD]: Oddly enough bazel produces a warning in my environment that `files` doesn't exist anymore (deprecated?) and that users should switch to `srcs`. This is consistent with bazel document here  https://docs.bazel.build/versions/master/be/pkg.html#pkg_tar.

Comment (ldemailly) [reply tools/deb/BUILD]: probably you are using a different bazel version than the prow one - for now prow wins (might be a great idea to make a new prow image, it's in test-infra / or ask the oncall)

**Comment by ldemailly** [comment]
@ayj it still doesn't pass the build

and needs release-note set to NONE (or filled; just not empty)

**Pull request closed without merging**